core/chain_manager_test.go

```diff
@@ -483,22 +485,71 @@ func TestInsertNonceError(t *testing.T) {
 	}
 }
 
-/*
-func TestGenesisMismatch(t *testing.T) {
-	db, _ := ethdb.NewMemDatabase()
-	var mux event.TypeMux
-	genesis := GenesisBlock(0, db)
-	_, err := NewChainManager(genesis, db, db, db, thePow(), &mux)
-	if err != nil {
+func TestChainReorgMissingTransactions(t *testing.T) {
```

Comment:
A more complete test imho would be to have three transactions in total. A, B included in the first chain, B, C included in the second chain. Then you could test all three reorg scenarios -> A drops, B stays, C appears